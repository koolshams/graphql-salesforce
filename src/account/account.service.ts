import { getOne, getSFConnection } from '../salesforce';

export interface Account {
  id: string;
  lang: string;
  countryCode: string;
  hasNewsletter: string;
  phoneNumber: string;
  countryIsoCode: string;
}

export const mapAccount = (row: any): Account => {
  return {
    id: row.Id,
    lang: row.Language__c,
    countryCode: row.Country__c,
    hasNewsletter: row.Has_Newsletter__c,
    phoneNumber: row.Phone,
    countryIsoCode: row.ShippingCountry
  };
};

export class AccountService {
  constructor(private connection: typeof getSFConnection) {}

  public async getAccount(accountId: string) {
    const sfConnection = await this.connection();
    const result = await sfConnection.query({
      query: `
      SELECT
        Id,
        Language__c,
        Country__c,
        Has_newsletter__c,
        Phone,
        Shippingcountry
      FROM
        Account
      WHERE
        Id = '${accountId}'
    `,
      raw: true
    });
    return getOne(result, mapAccount);
  }
}
