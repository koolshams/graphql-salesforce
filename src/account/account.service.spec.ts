import { mapAccount } from "./account.service";

test('mapper', () => {
    const m = mapAccount({});
    expect(m).toMatchSnapshot('accountMapper');
})