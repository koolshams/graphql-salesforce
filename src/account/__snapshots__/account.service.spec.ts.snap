// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapper: accountMapper 1`] = `
Object {
  "countryCode": undefined,
  "countryIsoCode": undefined,
  "hasNewsletter": undefined,
  "id": undefined,
  "lang": undefined,
  "phoneNumber": undefined,
}
`;
